app/pages/lab/feedback/types/common-form-hoc.jsx

```diff
@@ -0,0 +1,135 @@
+      },
+      failureEnabled: {
+        title: 'Enable negative feedback',
+        help: 'Shows a feedback message to the user when they incorrectly identify a known subject.'
+      },
+      defaultFailureMessage: {
+        title: 'Default failure message',
+        help: 'The message to show to the volunteer when they make an incorrect classification. Can be overridden using subject metadata.',
+      }
+    }
+  }
+});
+
+function returnCommonFormHOC(FeedbackTypeForm) {
+  return class CommonFormHOC extends React.Component {
+    constructor(props) {
+      super(props);
+      this.handleInputChange = this.handleInputChange.bind(this);
+      this.handleSave = this.handleSave.bind(this);
+      this.handleUpdateState = this.handleUpdateState.bind(this);
+      this.renderCommonFields = this.renderCommonFields.bind(this);
+      this.validate = this.validate.bind(this);
+      
+      const { feedback } = props;
+      this.state = {
+        form: {
+          id: feedback.id || '',
+          successEnabled: (typeof feedback.successEnabled === 'boolean') ? feedback.successEnabled : true,
+          defaultSuccessMessage: feedback.defaultSuccessMessage || '',
+          failureEnabled: (typeof feedback.failureEnabled === 'boolean') ? feedback.failureEnabled : true,
+          defaultFailureMessage: feedback.defaultFailureMessage || '',
+        },
+        validations: [
+          // Must have an id
+          (form) => (!!form.id),
+          // Must have either success or failure or both enabled
+          (form) => (!(!form.successEnabled && !form.failureEnabled)),
+          // Success must be disabled, or enabled with a message
+          (form) => (!form.successEnabled || (form.successEnabled && 
+            form.defaultSuccessMessage && form.defaultSuccessMessage.length > 0)),
+          // Failure must be disabled, or enabled with a message
+          (form) => (!form.failureEnabled || (form.failureEnabled && form.defaultFailureMessage && form.defaultFailureMessage.length > 0)),
+        ],
+      }
+    }
+
+    render() {
+      const newProps = {
+        formState: this.state.form,
+        updateState: this.handleUpdateState,
+      };
+
+      return (
+        <div className="edit-feedback-modal">
+          <Translate className="form-label" content="feedbackEditForm.title" />
+          {this.renderCommonFields()}
+          <FeedbackTypeForm {...this.props} {...newProps} />
+          <CommonFormHOCButtons 
+            disableSave={!this.state.form.valid} 
+            handleCancel={this.handleCancel} 
+            handleSave={this.handleSave} 
+          />
+        </div>
+      );
+    }
+
+    handleInputChange({ target }) {
+      this.handleUpdateState({
+        form: {
+          [target.name]: (target.type === 'checkbox') ? target.checked : target.value,
+        }
+      });
+    }
+
+    handleUpdateState(update) {
+      const newState = merge({}, this.state, update);
```

Comment:
Just clarifying what I think is going on here: lodash's merge is being used because a deep merge with state is happening and `setState`'s shallow merge is not sufficient?